added gazou package

jaminW55 opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Henry-Hiles reviewed (changes requested):
There are many issues with this which I have told to the author over discord. Some of these issues include:
- You haven't added yourself to the maintainers list file (should be separate commit before other one)
- Commit is named wrong
- Included a license and readme in the folder, which I believe is not normally done
- Didn't add package to all-packages.nix
- Package is in an bad location, as it is not a development tool.

NobbZ commented on pkgs/development/tools/gazou/LICENSE:
```diff
@@ -0,0 +1,674 @@
+    Copyright (C) <year>  <name of author>
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+Also add information on how to contact you by electronic and paper mail.
+
+  If the program does terminal interaction, make it output a short
+notice like this when it starts in an interactive mode:
+
+    <program>  Copyright (C) <year>  <name of author>
+    This program comes with ABSOLUTELY NO WARRANTY; for details type `show w'.
+    This is free software, and you are welcome to redistribute it
+    under certain conditions; type `show c' for details.
+
+The hypothetical commands `show w' and `show c' should show the appropriate
+parts of the General Public License.  Of course, your program's commands
+might be different; for a GUI interface, you would use an "about box".
+
+  You should also get your employer (if you work as a programmer) or school,
+if any, to sign a "copyright disclaimer" for the program, if necessary.
+For more information on this, and how to apply and follow the GNU GPL, see
+<https://www.gnu.org/licenses/>.
+
+  The GNU General Public License does not permit incorporating your program
+into proprietary programs.  If your program is a subroutine library, you
+may consider it more useful to permit linking proprietary applications with
+the library.  If this is what you want to do, use the GNU Lesser General
+Public License instead of this License.  But first, please read
+<https://www.gnu.org/licenses/why-not-lgpl.html>.
```
Delete this file please

jaminW55 replied:
Removed.

NobbZ commented on pkgs/development/tools/gazou/README.md:
```diff
@@ -0,0 +1,2 @@
+# gazou.nix
+NixOS adapted Gazou Package (https://github.com/kamui-fin/gazou)
```
Delete this as well please

jaminW55 replied:
Removed.

NobbZ commented on pkgs/development/tools/gazou/default.nix:
```diff
@@ -0,0 +1,28 @@
+
+stdenv.mkDerivation rec {
+  pname = "gazou";
+  version = "0.001-2023-08-18";
```
Latest release seems to be the 0.2.1, please use that instead

jaminW55 replied:
Correct Adjustment Verification:
version = "0.2.1";

jaminW55 replied:
Addressed.

NobbZ commented on pkgs/development/tools/gazou/default.nix:
```diff
@@ -0,0 +1,28 @@
+# gazou.nix
+{ stdenv, fetchFromGitHub, cmake, pkgconfig , lib , tesseract , leptonica , qt5 }:
+
+stdenv.mkDerivation rec {
```
Unnecessary `rec`

jaminW55 replied:
Removed.

NobbZ commented on pkgs/development/tools/gazou/default.nix:
```diff
@@ -0,0 +1,28 @@
+  src = fetchFromGitHub {
+    owner = "kamui-fin";
+    repo = "gazou";
+    rev = "2114963698c94ec7e8f9023f8b6dbfa8437337de";
```
Please use use `refs/tags/v0.2.1` here to match the version associated commit

jaminW55 replied:
Addressed.

NobbZ commented on pkgs/development/tools/gazou/default.nix:
```diff
@@ -0,0 +1,28 @@
+    owner = "kamui-fin";
+    repo = "gazou";
+    rev = "2114963698c94ec7e8f9023f8b6dbfa8437337de";
+    sha256 = "sha256-CiIQNwL+vrCLgmOdPt6OaS0dJfw46lhM9u4hUSlTW5I=";
```
Please use `hash` instead and update it accordingly as we changed `rev` as well.

jaminW55 replied:
Addressed.

NobbZ commented on pkgs/development/tools/gazou/default.nix:
```diff
@@ -0,0 +1,28 @@
+  meta = {
+    description = "A OCR for Japanese and Chinese Characters";
+    license = lib.licenses.gpl3Only;
+    maintainers = with stdenv.lib.maintainers; [ jaminW55 ];
```
The attribute `lib` in `stdenv` doesn't exist.

Please use `lib` directly.

Also `jaminW55` does not exist in `maintainers/maintainer-list.nix`, please add yourself there in a commit **before** referencing yourself here.

```suggestion
    maintainers = with lib.maintainers; [ jaminW55 ];
```

jaminW55 replied:
Added. Now to Commit.

NobbZ commented on pkgs/development/tools/gazou/default.nix:
```diff
@@ -0,0 +1,28 @@
+
+  meta = {
+    description = "A OCR for Japanese and Chinese Characters";
+    license = lib.licenses.gpl3Only;
```
Please als add a `homepage`, if there is none, please linkt the source repo.

jaminW55 replied:
Added.

jaminW55 committed:
maintainers: Add jaminW55 as a maintainer

maintainers: Add jaminW55 as a maintainer

maintainers: Add jaminW55 as a maintainer

jaminW55 committed:
gazou: init at 0.2.1

NobbZ commented on pkgs/tools/text/Gazou/default.nix:
```diff
@@ -0,0 +1,29 @@
+
+stdenv.mkDerivation {
+  pname = "gazou";
+  version = "unstable-2023-05-14";
```
As there are releases available, please add a comment explaining why you use a snapshot.

jaminW55 replied:
Current release does not build properly or have proper support for X11. Verified w/ some community members this was the case as well.

NobbZ replied:
I have seen the discussion in the chat, but not the poeple reading this drv in a week or month, please provide a comment, or add patches that would fix the issue for the release (with a comment).

Do the comment in the file, not here.

jaminW55 replied:
I can simply provide the comment note to the right of the used hash to indicate the reason?

NobbZ replied:
Personally I'd provide it close to the `version` rather than the `src`. As the `src` just follows the `version` usually.

NobbZ reviewed (changes requested):
You are still missing the entry in the top-level

FliegendeWurst commented on pkgs/tools/text/Gazou/default.nix:
New packages go in pkgs/by-name (new policy)
